packages/edit-post/src/components/layout/index.native.js

```diff
@@ -157,12 +160,14 @@ class Layout extends Component {
 
 export default compose( [
 	withSelect( ( select ) => {
-		const { __unstableIsEditorReady: isEditorReady } = select(
-			'core/editor'
-		);
+		const {
+			__unstableIsEditorReady: isEditorReady,
+			getCurrentPostType,
+		} = select( 'core/editor' );
 		const { getEditorMode } = select( 'core/edit-post' );
 
 		return {
+			isPage: getCurrentPostType(),
```

Comment:
I think this should be `getCurrentPostType() === 'page'`.

I think I'd still encapsulate that logic in the HOC, maybe splitting the `PageTemplatePickerVisible` HOC/hook into two: "available" (is a page) and "visible" (has no content).